src/main/java/org/fcrepo/auth/webac/WebACRolesProvider.java

```diff
@@ -389,14 +393,24 @@ private FedoraResource getBaseVersion(final FedoraResource resource) {
                     final Map<String, List<String>> aclTriples = new HashMap<>();
                     child.getTriples(translator, PROPERTIES)
                         .filter(p -> isAclPredicate.test(model.asStatement(p).getPredicate()))
-                        .forEach(t -> {
-                            aclTriples.putIfAbsent(t.getPredicate().getURI(), new ArrayList<>());
-                            if (t.getObject().isURI()) {
-                                aclTriples.get(t.getPredicate().getURI()).add(
-                                    substringBeforeLast(t.getObject().getURI(), "/" + JCR_CONTENT));
-                            } else if (t.getObject().isLiteral()) {
-                                aclTriples.get(t.getPredicate().getURI()).add(
-                                    t.getObject().getLiteralValue().toString());
+                        .forEach(triple -> {
+                            final String predicate = triple.getPredicate().getURI();
+                            aclTriples.putIfAbsent(predicate, new ArrayList<>());
+                            if (triple.getObject().isURI()) {
+                                final String object = triple.getObject().getURI();
+                                if (predicate.equals(WEBAC_AGENT_VALUE)) {
+                                    if (userBaseUri != null && object.startsWith(userBaseUri)) {
+                                        aclTriples.get(predicate).add(object.substring(userBaseUri.length()));
```

Comment:
Yes, that case is covered with line 408. Here is the issue -- there are two ways to get a user id into the webac module: via Servlet authentication in which case, it will be a bare String and will need to be prefixed or via header (i.e. delegated AuthN) in which case the client has full control over whether the value is a simple userID string (user1, user2, etc) or a full URI (info:user/user1, info:user/user2, etc).

For the first case, where we need to transform the `Resource`-based URI into a base String, via the prefix. In the second case, there is no such need. It is also possible that a user with authenticate via Servlet authN and then rely on delegated authN, so it is entirely possible that both cases need to be supported simultaneously -- that is, where `info:user/user1` will need to be transformed to `user1` and `info:user/user5` needs to stay in that form.